ateam_radio_bridge/src/conversion.cpp

```diff
@@ -64,4 +64,75 @@ ateam_msgs::msg::RobotFeedback Convert(const BasicTelemetry & basic_telemetry)
+  ateam_motor_feedback.velocity = motor_debug_telemetry.wheel_velocity;
+  ateam_motor_feedback.torque = motor_debug_telemetry.wheel_torque;
+}
+
+ateam_msgs::msg::RobotMotorFeedback ConvertMotorDebugTelemetry(const MotorDebugTelemetry & motor_debug_telemetry) {
+  ateam_msgs::msg::RobotMotorFeedback robot_motor_feedback;
+
+  ConvertMotorDebugTelemetry(motor_debug_telemetry, robot_motor_feedback);
+
+  return robot_motor_feedback;
+}
+
+ateam_msgs::msg::RobotMotionFeedback ConvertControlDebugTelemetry(const ControlDebugTelemetry & control_debug_telemetry) {
+  ateam_msgs::msg::RobotMotionFeedback robot_motion_feedback;
+
+  robot_motion_feedback.motors[robot_motion_feedback.FRONT_LEFT_MOTOR] = ConvertMotorDebugTelemetry(control_debug_telemetry.motor_fl);
+  robot_motion_feedback.motors[robot_motion_feedback.FRONT_RIGHT_MOTOR] = ConvertMotorDebugTelemetry(control_debug_telemetry.motor_fr);
+  robot_motion_feedback.motors[robot_motion_feedback.BACK_RIGHT_MOTOR] = ConvertMotorDebugTelemetry(control_debug_telemetry.motor_br);
+  robot_motion_feedback.motors[robot_motion_feedback.BACK_LEFT_MOTOR] = ConvertMotorDebugTelemetry(control_debug_telemetry.motor_bl);
+
+  robot_motion_feedback.imu.orientation_covariance[0] = -1.0;  // ROS2 docs say if a sensor doesn't provide a data point, then set element '0' of it's covariance to
+  robot_motion_feedback.imu.angular_velocity = ConvertFloatArrayToVec3(control_debug_telemetry.imu_gyro);
+  robot_motion_feedback.imu.linear_acceleration = ConvertFloatArrayToVec3(control_debug_telemetry.imu_accel);
+
+  robot_motion_feedback.body_velocity_setpoint = ConvertFloatArrayToTwist(control_debug_telemetry.commanded_body_velocity);
+  robot_motion_feedback.clamped_body_velocity_setpoint = ConvertFloatArrayToTwist(control_debug_telemetry.clamped_commanded_body_velocity);
+  robot_motion_feedback.body_velocity_state_estimate = ConvertFloatArrayToTwist(control_debug_telemetry.cgkf_body_velocity_state_estimate);
+  robot_motion_feedback.body_velocity_control_variable = ConvertFloatArrayToTwist(control_debug_telemetry.body_velocity_u);
+
+  // copying from some C arrays here, not a struct so add some sanity checks
+  assert((sizeof(control_debug_telemetry.wheel_velocity_u) / sizeof(control_debug_telemetry.wheel_velocity_u[0]) <= robot_motion_feedback.wheel_velocity_control_variable.size()));
+  assert((sizeof(control_debug_telemetry.wheel_velocity_clamped_u) / sizeof(control_debug_telemetry.wheel_velocity_clamped_u[0]) <= robot_motion_feedback.clamped_wheel_velocity_control_variable.size()));
+  std::copy(std::begin(control_debug_telemetry.wheel_velocity_u), std::end(control_debug_telemetry.wheel_velocity_u), std::begin(robot_motion_feedback.wheel_velocity_control_variable));
+  std::copy(std::begin(control_debug_telemetry.wheel_velocity_clamped_u), std::end(control_debug_telemetry.wheel_velocity_clamped_u), std::begin(robot_motion_feedback.clamped_wheel_velocity_control_variable));
```

Comment:
A more modern approach to this kind of thing uses [std::span](https://en.cppreference.com/w/cpp/container/span) to wrap the raw array and [std::ranges::copy](https://en.cppreference.com/w/cpp/algorithm/ranges/copy) for a shorter call to copy the whole container.

```suggestion
  std::span wheel_velocity_u_span{control_debug_telemetry.wheel_velocity_u};
  static_assert(wheel_velocity_u_span.size() == robot_motion_feedback.wheel_velocity_control_variable.size());
  std::ranges::copy(wheel_velocity_u_span, robot_motion_feedback.wheel_velocity_control_variable.begin());

  std::span wheel_velocity_clamped_u_span{control_debug_telemetry.wheel_velocity_clamped_u};
  static_assert(wheel_velocity_clamped_u_span.size() == robot_motion_feedback.clamped_wheel_velocity_control_variable.size());
  std::ranges::copy(wheel_velocity_clamped_u_span, robot_motion_feedback.clamped_wheel_velocity_control_variable.begin());
```